core/src/main/java/org/jdbi/v3/core/statement/DefineNamedBindingMode.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.statement;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.NullArgument;
+import org.jdbi.v3.meta.Beta;
+
+/**
+ * Selects the mode of operation of the statement {@code defineNamedBindings} feature.
+ */
+@Beta
+public enum DefineNamedBindingMode {
+    /**
+     * Bind a boolean {@code true} for present values, {@code false} for null values.
+     * Default behavior.
+     *
+     * Note that due to implementation limitations, null arguments must be instances of
+     * {@link org.jdbi.v3.core.argument.NullArgument} to be recognized.
+     */
+    BOOLEAN {
+        @Override
+        Optional<Object> apply(Argument arg) {
+            return Optional.of(arg instanceof NullArgument ? false : true);
+        }
+    },
+    /**
+     * Bind the Argument's {@code toString} value as the template parameter.
+     * Please be cautious not to allow SQL injection!
+     */
+    TO_STRING {
```

Comment:
The eternal plague of existing code influencing the evolution of new code, instead of making new things from scratch without bias from outside...

Reply:
Hm, I guess that works.  I don't disagree with your reasoning however there is only so far we can go to protect users from themselves.  `define` in general is suspect to this sort of issue and this isn't something new, you can get yourself in trouble just the same with defining any `String` values at all.

Reply:
And we (almost) have to make them `enum`s -- otherwise you can't attach them to the annotation.  Unfortunately it's not possible to send e.g. an arbitrary Function in.  We'd have to take a `Class` and specify it has a nullary constructor, etc, much like `JdbiConfig`, which is also a big bummer.  Ends up being a fair amount of boilerplace to support something that almost nobody will extend.  I'm not necessarily against changing it, just pointing out that there is a nontrivial reason we went with the `enum` type, and if we are going to object to the only other implementation maybe I should just make it hard-coded for now until anyone actually asks for it...

Reply:
I've dropped the customizable implementation, now it only defines booleans for now.  We can always add it back later when we understand its uses better.  #YAGNI